spec/app/controllers/courses_controller/controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe CoursesController, type: :request do
+  let(:request_payload) { 
+    {
+      course_uuid: given_course_data.uuid,
+      ecosystem_uuid: given_course_data.ecosystem_uuid
+    }
+  }
+
+  let(:given_course_data) {
+    build(:course, ecosystem_uuid: ecosystem.uuid)
+  }
+
+  let(:ecosystem) {
+    build(:ecosystem)
+  }
+
+  let(:target_result) {
+    {
+      created_course_uuid: given_course_data.uuid
+    }
+  }
+
+  let(:service_double) {
+    dbl = object_double(Services::CreateCourse::Service.new)
+    allow(dbl).to receive(:process)
+              .with(**request_payload)
```

Comment:
This is interesting.  I don't _hate_ it, but it suggests to the reader that somehow the parameters to the service _must_ come from the `:request_payload` (which _happens_ to be true here, but is not in general).  IMO, being more explicit here would reduce confusion, especially if the schema changes by the service doesn't:
```ruby
   allow(dbl).to receive(:process)
              .with(
                course_uuid:    given_course_uuid,
                ecosystem_uuid: given_ecosystem_uuid,
              ).and_return(target_result)
```